### PR TITLE
Rename transition constraints to integrity constraints

### PR DESCRIPTION
**Pull request by tohrnii**
Partly addressing #84.

This PR renames transition constraints to integrity constraints to make the section name more generic to be able to differentiate between validity and transition constraints in the IR.

### CODE REVIEW

File: ir/src/lib.rs
Comment (grjte) [inline ir/src/lib.rs:196]: I'm not sure about the name `integrity_graph`. The rest of the renaming mostly makes sense and preserves meaning, but this feels like the meaning changes a bit. I think it might be better to call this `constraint_graph`. What do you think? I know it doesn't include the boundary constraints, but I think that's ok (and at some point in the future it might).

File: codegen/winterfell/src/air/mod.rs
Comment (grjte) [inline codegen/winterfell/src/air/mod.rs:17]: I think it might be better not to rename these, since Winterfell doesn't have an idea of `integrity constraints`, it only has `boundary` or `transition` constraints. I think that our codegen crates will probably have the most clarity if they generally use the same language as the target. This is just my opinion though. What do you think?

If we agree on this, we should restore the comments below and the text in the README as well

File: ir/src/lib.rs
Comment (grjte) [inline ir/src/lib.rs:15]: I think we should probably rename `TransitionConstraintDegree` as well, to keep everything in the IR consistent. That would mean updating all of the related comments, etc. as well. It seems like a deliberate decision that you chose not to do this though, so it would be good to understand your reasoning & discuss

**Review by grjte — CHANGES_REQUESTED** [review]
Thank you! This looks good to me overall, but I left a few comments inline for discussion.

File: ir/src/lib.rs
Comment (tohrnii) [reply ir/src/lib.rs:15]: I wasn't sure if we wanted to change this naming since this struct is defined in winterfell so I wasn't sure if we should keep it the same in IR as well or not. But I think it makes sense to change this in IR. Thanks.

File: codegen/winterfell/src/air/mod.rs
Comment (tohrnii) [reply codegen/winterfell/src/air/mod.rs:17]: Makes sense. Will revert. Thanks.

File: ir/src/lib.rs
Comment (grjte) [reply ir/src/lib.rs:196]: I would do `constraint_graph`

File: ir/src/lib.rs
Comment (grjte) [reply ir/src/lib.rs:15]: Cool, I figured that was your thinking (and it makes sense). Sounds good to change it for our IR though

File: ir/src/lib.rs
Comment (bobbinth) [reply ir/src/lib.rs:196]: I like `constraint_graph`. I actually think that in the future we'll probably fold boundary constraints into this graph as well.

**Comment by tohrnii** [comment]
Thank you @grjte for the review. Made suggested changes. I'll rebase this one after 102 is merged and we can merge this one after that.

File: ir/README.md
Comment (Al-Kindi-0) [inline ir/README.md:28]: nit: is the sentence "...to a size..." complete?

**Review by Al-Kindi-0 — COMMENTED** [review]
Looks good to me, thank you! Only one minor nit which might not be relevant.

File: ir/README.md
Comment (tohrnii) [reply ir/README.md:28]: @Al-Kindi-0 It seems complete to me. Not sure if I'm missing something.

Comment (grjte) [reply ir/README.md:28]: Technically it's not complete, because we have a comma instead of a verb (instead of "Public inputs are represented by a vector that maps..."). That's because it's really a continuation of the sentence above the set of bullet points. It might be clearer to say "Public inputs, which are represented by...", in which case the other bullet points should be similarly updated.

Comment (grjte) [reply ir/README.md:28]: But it's technically correct as it is, as long as the whole chunk is read together (it's just not a sentence on its own)

**Review by grjte — APPROVED** [review]
Looks good, thanks @tohrnii!

**Review by Al-Kindi-0 — APPROVED** [review]
Looks good, thanks @tohrnii !

**Review by Overcastan — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**
